agent/consul/server.go

```diff
@@ -467,10 +471,19 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server) (*Ser
 	})
 
 	// TODO(NET-1380, NET-1381): thread this into the net/rpc and gRPC interceptors.
-	s.incomingRPCLimiter = rpcRate.NewHandler(rpcRate.HandlerConfig{
-		// TODO(NET-1379): pass in _real_ configuration.
-		GlobalMode: rpcRate.ModePermissive,
-	}, s)
+	if s.incomingRPCLimiter == nil {
+		s.incomingRPCLimiter = rpcRate.NewHandler(rpcRate.HandlerConfig{
+			GlobalMode: rpcRate.Mode(config.RequestLimitsMode),
+			GlobalReadConfig: multilimiter.LimiterConfig{
+				Rate:  config.RequestLimitsReadRate,
+				Burst: int(config.RequestLimitsReadRate) * requestLimitsBurstMultiplier,
+			},
+			GlobalWriteConfig: multilimiter.LimiterConfig{
+				Rate:  config.RequestLimitsWriteRate,
+				Burst: int(config.RequestLimitsWriteRate) * requestLimitsBurstMultiplier,
+			},
```

Comment:
Maybe later when we get to IP rate limiting, but I think this should be a substruct so we can use it for other kind of limiting.

Reply:
This has been fixed.  Thank you for the suggestion!